jwst/datamodels/tests/test_model_container.py

```diff
@@ -95,18 +96,6 @@ def test_modelcontainer_error_from_asn(tmp_path):
         datamodels.open(path)
 
 
-@pytest.mark.parametrize("model", [v for v in defined_models.values()])
```

Comment:
@braingram commented on the JIRA ticket:

I would vote for deleting test_all_datamodels_init. It is effectively a duplicate of a test in stdatamodels:
[see here](https://github.com/spacetelescope/stdatamodels/blob/e91f81f037029a7d58ee8f822cef61ad1d789de9/src/stdatamodels/jwst/datamodels/tests/test_models.py#L313)
It is parametrized with the same values (which do not include SourceModelContainer so the conditional that differs in the jwst test is never visited). It is likely this test could have been removed in [this PR](https://github.com/spacetelescope/jwst/pull/7439)

Reply:
It came up because this test was showing deprecation warnings, e.g., 
DeprecationWarning: "MIRIRampModel" is deprecated and will be removed. Use RampModel.

Reply:
It's been a while since we deprecated those data models. Perhaps it's time to remove them.

Reply:
The same situation also exists for `DrizProductModel` and `MultiProductModel`.  I will go ahead and remove all three as part of this PR unless someone feels strongly that this is a bad idea

Reply:
Just made a new ticket for this.  Most of the changes would be in stdatamodels anyway. [JP-3585](https://jira.stsci.edu/browse/JP-3585)

Reply:
xref: https://github.com/spacetelescope/stdatamodels/pull/171